### PR TITLE
HaikuOS port

### PR DESCRIPTION
**Pull request by SanyaSho**

### CODE REVIEW

File: thirdparty/raknet/SocketLayer.cpp [hunk 1]
```diff
@@ -24,6 +24,10 @@
 #include <netdb.h>
 #endif
 
+#if defined( __HAIKU__ )
```
Comment (TheBrokenRail) [inline thirdparty/raknet/SocketLayer.cpp:27]:
```suggestion
#ifdef __HAIKU__
```

File: platforms/sdl/CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,11 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     add_compile_options(-Wno-inconsistent-missing-override -Wno-enum-compare-switch -Wno-register)
 endif()
 
+# Network library
+if(HAIKU)
+    add_link_options(-lnetwork)
```
Comment (TheBrokenRail) [inline platforms/sdl/CMakeLists.txt]:
```suggestion
    link_libraries(network)
```

File: source/world/level/storage/ExternalFileLevelStorageSource.cpp [hunk 3]
```diff
@@ -58,7 +58,16 @@ void ExternalFileLevelStorageSource::getLevelList(std::vector<LevelSummary>& vls
 
 		LOG_I("Entry: %s", de->d_name);
 
+#if defined( __HAIKU__ )
+		// @HACK: make a temp buffer here
+		char tempbuf[PATH_MAX];
+		snprintf( tempbuf, sizeof( tempbuf ), "%s/%s", m_worldsPath.c_str(), de->d_name );
+
+		struct stat buf;
+		if ( ( lstat( tempbuf, &buf ) == 0 ) && S_ISDIR( buf.st_mode ) )
```
Comment (TheBrokenRail) [inline source/world/level/storage/ExternalFileLevelStorageSource.cpp]:
```suggestion
		std::string temp = m_worldsPath + '/' + de->d_name;
	
		struct stat buf;
		if ( ( lstat( temp.c_str(), &buf ) == 0 ) && S_ISDIR( buf.st_mode ) )
```

**Review by TheBrokenRail — CHANGES_REQUESTED** [review]
Just some small suggestions.

**Comment by iProgramMC** [comment]
I agree with TheBrokenRail's suggestions.

File: thirdparty/raknet/SocketLayer.cpp (hunk 1 above)
Comment (SanyaSho) [reply thirdparty/raknet/SocketLayer.cpp:27]: there's no use change it because raknet contains both ifdefs and if defineds

**Pull request merged**